pkg/gateway/model_build_targetgroup.go

```diff
@@ -192,23 +193,39 @@ func (t *latticeServiceModelBuildTask) buildTargets(ctx context.Context) error {
 // Build target group for K8S serviceexport object
```

Comment:
Reviewers could check the PR description first before review this code.

To review the model_build_targetgroup.go code, checking the code "diff" is hard to review. 
Reviewers could:
- Open one browser tab for the OLD code: https://github.com/aws/aws-application-networking-k8s/blob/f5b903321a1fc226e5bda3e2a0c2bfc06dd1343d/pkg/gateway/model_build_targetgroup.go#L193-L273
- Open another browser tab for the NEW code https://github.com/zijun726911/aws-application-networking-k8s/blob/77efd4bf55b9b04ef7e24d9c749d1c5fd67d6efa/pkg/gateway/model_build_targetgroup.go#L213-L284
- Compare them side by side. I think new code is clear and easy to review.